src/bin/miri.rs

```diff
@@ -98,10 +98,9 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
             }
 
             if tcx.sess.opts.optimize != OptLevel::No {
-                tcx.dcx().warn("Miri does not support optimizations. If you have enabled optimizations \
-                    by selecting a Cargo profile (such as --release) which changes other profile settings \
-                    such as whether debug assertions and overflow checks are enabled, those settings are \
-                    still applied.");
+                tcx.dcx().warn("Miri does not support optimizations: the `-O` flag is ignored. The only effect \
```

Comment:
Why is it useful to mention -O? I've only seen that flag used internally, mostly in codegen tests.

Reply:
All right, it's "opt-level" now.